---
title: "RBRA_data_analysis_proj2_Q13921"
author: "Pieter van Veelen"
date: "7/27/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

options(scipen = 999) # tell R to avoid all scientific notation

setwd("C:/Users/Rita/Documents/R/Article1/scripts")

# Packages names
packages <- c("phyloseq", "qiime2R", "tidyverse", "magrittr", "devtools", "qiime2R", "here", "breakaway", "DivNet", "openxlsx", "ape", "vegan", "ggtext", "cowplot", "RColorBrewer", "microbiome", "lme4", "lmerTest","decontam", "remotes", "ampvis2", "speedyseq")

# Install packages not yet installed
installed_packages <- packages %in% rownames(installed.packages())
if (any(installed_packages == FALSE)) {
  install.packages(packages[!installed_packages])
}

# Packages not available in this version of R - 4.1.2
## phyloseq
#if(!requireNamespace("BiocManager")){install.packages("BiocManager")}
#BiocManager::install("phyloseq")

## qiime2R
#if (!requireNamespace("devtools", quietly = TRUE)){install.packages("devtools")}
#devtools::install_github("jbisanz/qiime2R")

## breakaway
#install.packages("devtools")
#devtools::install_github("adw96/breakaway")

## DivNet
#remotes::install_github("adw96/breakaway")
#remotes::install_github("adw96/DivNet")

## microbiome
#library(BiocManager)
#BiocManager::install("microbiome")

## decontam
#if (!requireNamespace("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")
#BiocManager::install("decontam")

## ampvis2
#if (!requireNamespace("ampvis2", quietly = TRUE))
    #install.packages("ampvis2")
#remotes::install_github("MadsAlbertsen/ampvis2", Ncpus = 6)
#https://sites.google.com/a/ciad.mx/bioinformatica/home/metagenomica/visualizacion/ampvis2 and https://madsalbertsen.github.io/ampvis2/articles/ampvis2.html

## speedyseq
#if (!requireNamespace("speedyseq", quietly = TRUE))
    #install.packages("speedyseq")
#remotes::install_github("mikemc/speedyseq")
```

## Exploratory data analysis

This is an R Markdown document contains an analysis of microbial community dynamics of an experiment of Rita Branco and Koen Deurloo in collaboration with Pieter van Veelen. 

### Loading libraries

```{r library loading, message=F, echo=T, eval=T, warning=T, include=F, cache=F}
## load required packages
library(phyloseq)
library(qiime2R)
library(tidyverse)
library(magrittr)
library(devtools)
library(qiime2R)
library(here)
library(breakaway)
library(DivNet)
library(openxlsx)
library(ape)
library(vegan)
library(ggtext)
library(cowplot)
library(RColorBrewer)
library(microbiome)
library(lme4)
library(lmerTest)
library(decontam)
library(remotes)
library(ampvis2)
library(speedyseq)
```


### Data import

Below, the input data are imported in R. These files are output files from our QIIME2 (v.2019.10) sequence data pre-processing workflow. Scripts and parameter settings are found in separate bash files.

```{r import data, message=F, echo=T, eval=T, warning=T, include=F, cache=F}
#creating phyloseq objects with 
physeq = qza_to_phyloseq(features = "C:/Users/Rita/Documents/R/Article1/input_data/RBRA_proj2_Q13921_16S_515F926R_20210701_table.qza",
                        tree = "C:/Users/Rita/Documents/R/Article1/input_data/RBRA_proj2_Q13921_16S_515F926R_20210701_rooted-tree.qza",
                        taxonomy = "C:/Users/Rita/Documents/R/Article1/input_data/RBRA_proj2_Q13921_16S_515F926R_20210701_taxonomy_NB_classifier_SILVA_132_99_16S_515F-926R_QIIME2-2019.10.qza",
                        metadata = "C:/Users/Rita/Documents/R/Article1/input_data/RBRA_proj2_Q13921_16S_515F926R_20210701@metadata_formatted_v2.txt")

colnames(sample_data(physeq))
physeq

```

### Cleaning data set

The following steps are subsequently performed to clean the data: 1) bifurcating tree using ape package; 2) cleaning up the metadata; 3) replacing taxonomic strings that are empty, NA, metagenome, ambiguous taxa; 4) decontaminating based on frequency and prevalence using the DECONTAM package; 5) split blanks from samples; 

#### 1) Resolve phylogenetic tree

Number of tips = number internal nodes + 1. If this does not happens, then nodes from the phylogenetic tree have to be splited/bifurcated.

```{r clean phylogeny, message=F, echo=F, eval=T, warning=T, include=F, cache=F}
# evaluate tree topology
is.binary(phy_tree(physeq)) # if FALSE --> polychotomy present (node has more than 2 tips)

# if FALSE:
# resolve polychotomous nodes
phy_tree_resolved <- multi2di(phy_tree(physeq))
is.binary(phy_tree_resolved)

# create new phy_tree
tree2 <- phy_tree_resolved

# subset_taxa(phy_tree_resolved, Kingdom ==  "Bacteria")

# merge new phy_tree object with sample_data and otu_table into new phyloseq object
psdata_RBRA <- merge_phyloseq(otu_table(physeq), sample_data(physeq), tax_table(physeq), tree2)
psdata_RBRA

class(sample_data(psdata_RBRA)$Sample_type)
```

#### 2) Clean-up metadata

```{r}
sample_data(psdata_RBRA)$Redox_type <- factor(sample_data(psdata_RBRA)$Redox_type,
                                              levels=c("inoculum", "aerobic", "nitrate", "iron", "sulphate", "methanogenic"))

sample_data(psdata_RBRA)$Sample_type <- factor(sample_data(psdata_RBRA)$Sample_type, levels=c("soil", "mun", "ditch", "ind", "blank"))

names(sample_data(psdata_RBRA)) <- tolower(names(sample_data(psdata_RBRA))) # to avoid mistakes typing in both upper and lower case

```

#### 3) Clean-up taxa data

```{r clean taxanomy, message=F, echo=F, eval=T, warning=T, include=F, cache=F}
## remove Archaea, Chloroplast and Mitochondia and unassigned Kingdom

#View(tax_table(psdata_RBRA)) # 11217 ASVs in total
#physeq1 <- subset_taxa(psdata_RBRA, Kingdom == "D_0__Bacteria") # retain Archaea
#physeq2 <- subset_taxa(psdata_RBRA, Phylum == "Chloroplast") # none found
#physeq3 <- subset_taxa(psdata_RBRA, Class == "Chloroplast") # none found
physeq1 <- subset_taxa(psdata_RBRA, Order != "Chloroplast") # dropped 635 ASVs
physeq2 <- subset_taxa(physeq1, Family != "Mitochondria") # dropped 325 ASVs

## clean taxonomy tags with no information
# specify NA taxon name tags to last known taxon names
tax.clean <- data.frame(tax_table(physeq2))
#tax.clean <- data.frame(tax.clean)
mode(tax.clean)
str(tax.clean)

# change tags into characters - required for further code
tax.clean2 <- tax.clean
tax.clean2$Kingdom <- as.character(tax.clean2$Kingdom)
tax.clean2$Phylum <- as.character(tax.clean2$Phylum)
tax.clean2$Class <- as.character(tax.clean2$Class)
tax.clean2$Order <- as.character(tax.clean2$Order)
tax.clean2$Family <- as.character(tax.clean2$Family)
tax.clean2$Genus <- as.character(tax.clean2$Genus)
tax.clean2$Species <- as.character(tax.clean2$Species)
str(tax.clean2)

# remove ambiguous taxa
tax.clean = tax.clean2 %>% mutate_all(funs(str_replace(., "Ambiguous_taxa", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "metagenome", "")))
#tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "uncultured archaeon", "")))
#tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "uncultured bacterium", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "uncultured", "")))
#tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "unknown", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "unidentified", "")))
tax.clean[is.na(tax.clean)] <- ""

# function replace name tags from [https://github.com/joey711/phyloseq/issues/850]
for (i in 1:nrow(tax.clean)){
  if (tax.clean[i,2] == ""){
    Kingdom <- paste("Kingdom_", tax.clean[i,1], sep = "")
    tax.clean[i, 2:7] <- Kingdom
  } else if (tax.clean[i,3] == ""){
    Phylum <- paste("Phylum_", tax.clean[i,2], sep = "")
    tax.clean[i, 3:7] <- Phylum
  } else if (tax.clean[i,4] == ""){
    Class <- paste("Class_", tax.clean[i,3], sep = "")
    tax.clean[i, 4:7] <- Class
  } else if (tax.clean[i,5] == ""){
    Order <- paste("Order_", tax.clean[i,4], sep = "")
    tax.clean[i, 5:7] <- Order
  } else if (tax.clean[i,6] == ""){
    Family <- paste("Family_", tax.clean[i,5], sep = "")
    tax.clean[i, 6:7] <- Family
  } else if (tax.clean[i,7] == ""){
    tax.clean$Species[i] <- paste("Genus_",tax.clean$Genus[i], sep = "_")
  }
}

psdata_newTaxtab <- psdata_RBRA
tax_table(psdata_newTaxtab) <- as.matrix(tax.clean)
tax_table(psdata_newTaxtab) #save previous tax_table not to overwrite

# put cleaned tax_table into phyloseq object
tax_table(physeq2) <- as.matrix(tax.clean)

psdata_RBRA <- physeq2 # overwrite psdata_RBRA for cleaned data

```

#### 4) Remove contaminants with DECONTAM

```{r decontam, message=F, echo=T, eval=T, warning=T, include=T, cache=F}

# add sample_or_control column
sample_data(psdata_RBRA)$is.neg <- sample_data(psdata_RBRA)$sample_type == "blank"
sample_data(psdata_RBRA)$sample_or_control <- factor(if_else(sample_data(psdata_RBRA)$sample_type == "blank", "control", "sample"))

# replace DNA concentration = 0 for a very low DNA amount (necessary for further steps, otherwise samples is taken out of analysis)
sample_data(psdata_RBRA)$dna_ng_ml <- if_else(sample_data(psdata_RBRA)$dna_ng_ml == 0, 
                                              sample_data(psdata_RBRA)$dna_ng_ml+0.0001, 
                                              sample_data(psdata_RBRA)$dna_ng_ml)
  
df <- as.data.frame(sample_data(psdata_RBRA)) # put sample_data into a ggplot-friendly data.frame
df$librarysize <- sample_sums(psdata_RBRA) # 'samples_sums' returns the total number of individuals observed from each sample
df <- df[order(df$librarysize),] # 'order' returns a permutation which rearranges its first argument into ascending or descending order
df$index <- seq(nrow(df)) # 'seq' generates regular sequences, 'nrow' return the number of rows or columns present in an array-like object

df %>%    
ggplot(aes(x=index, y=librarysize, color=sample_or_control)) + 
  geom_point() + 
  scale_y_continuous(trans = "log10") +
  labs(y = "Library size (log10)", x = "Sample rank") +
  theme_classic()

# Useful info 'decontam'[http://bioconductor.riken.jp/packages/3.7/bioc/vignettes/decontam/inst/doc/decontam_intro.html] and [https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-018-0605-2]

# contaminants based on frequency (probability as function of DNA concentration)
contamdf.freq <- isContaminant(psdata_RBRA, method="frequency", conc="dna_ng_ml")
contamdf.freq %>% arrange(p.freq, desc = F) %>% head()

table(contamdf.freq$contaminant) # 158 contaminants based on frequency, 'table' uses the cross-classifying factors to build a contingency table of the counts at each combination of factor levels
head(which(contamdf.freq$contaminant)) #  179 258 349 354 355 431, 'which' gives the TRUE indices of a logical object

plot_frequency(psdata_RBRA, taxa_names(psdata_RBRA)[c(1,431)], conc="dna_ng_ml") + 
  xlab("DNA Concentration (ng/ml)\n(Quant-IT dsDNA kit)")

plot_frequency(psdata_RBRA, taxa_names(psdata_RBRA)[sample(which(contamdf.freq$contaminant),3)], conc="dna_ng_ml") +
  xlab("DNA Concentration (ng/ml)\n(Quant-IT dsDNA kit)")

# contaminants based on prevalence
contamdf.prev <- isContaminant(psdata_RBRA, method="prevalence", neg="is.neg", threshold=0.1)
table(contamdf.prev$contaminant) # 2 contaminants based on prevalence
head(which(contamdf.prev$contaminant)) # 2657 6535

# Make phyloseq object of presence-absence in negative controls and true samples
ps.pa <- transform_sample_counts(psdata_RBRA, function(abund) 1*(abund>0)) # 'transform_sample_counts' transforms the sample counts of a taxa abundance matrix according to a user-provided function
ps.pa.neg <- prune_samples(sample_data(ps.pa)$sample_or_control == "control", ps.pa) # 'prune_samples' defines a subset of samples to keep in a phyloseq object
ps.pa.pos <- prune_samples(sample_data(ps.pa)$sample_or_control == "sample", ps.pa)

# Make dataframe of prevalence in positive and negative samples
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                      contaminant=contamdf.prev$contaminant) # 'taxa_sums' returns the total number of individuals observed from each species/taxa/OTU
ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
  xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)")

# contaminants based on prevalence and frequency
contamdf.comb <- isContaminant(psdata_RBRA, method="combined", neg="is.neg", conc = "dna_ng_ml")
table(contamdf.comb$contaminant) # 44 contaminants based on prevalence and frequency combined
contamdf.comb %>% arrange(p, desc = F) %>% filter(p < 0.1)

# saving contaminant and decontaminated data
#selected method: prevalence (low biomass samples)
psdata_contam <-  prune_taxa(contamdf.prev$contaminant, psdata_RBRA) # save contaminant psdata
psdata_decontam <- prune_taxa(!contamdf.prev$contaminant, psdata_RBRA) # save decontaminated psdata

```

#### Remove blanks from the dataset

```{r select RBRA samples, message=F, echo=T, eval=T, warning=T, include=T, cache=F}

## full dataset = psdata_decontam

# add SampleID
sample_data(psdata_decontam)$sampleid <- sample_names(psdata_decontam) # 'sample_names' gets sample names

# blanks, keep the taxa that are present on one or more of the blanks
psdata_RBRA_blank <- 
  prune_taxa(
    taxa_sums(
      subset_samples(psdata_decontam, sample_type == "blank")) > 0, 
      subset_samples(psdata_decontam, sample_type == "blank")) # 'subset_samples' subsets samples by sample_data expression

# samples, keep the taxa that are present on one ore more of the samples
psdata_RBRA_only <- 
  prune_taxa(
    taxa_sums(
      subset_samples(psdata_decontam, sample_type != "blank")) > 0, 
      subset_samples(psdata_decontam, sample_type != "blank"))

psdata_RBRA <- prune_taxa(taxa_sums(psdata_RBRA_only) >0, psdata_RBRA_only) # overwrite psdata_RBRA for data without blanks

```

### Rarefaction curves

#### calculate rarefaction curve 
```{r filter on abundance, message=F, echo=T, eval=T, warning=T, include=T, cache=F}

# relative abundance data
psdata_RBRA_rel <- transform_sample_counts(psdata_RBRA, fun = function(x) x/sum(x)) # 10177 taxa (100% abundance)

# total reads left = 3645984
sum(sample_sums(psdata_RBRA))

# abundance filter at (0.01%, 0.1%, 0.5%), keep above 99% abundance when removing rare data
psdata_RBRA_0.01pct <- prune_taxa(taxa_sums(psdata_RBRA_rel) > 0.0001, psdata_RBRA) # 7590 taxa (99.46% abundance)
psdata_RBRA_0.1pct <- prune_taxa(taxa_sums(psdata_RBRA_rel) > 0.001, psdata_RBRA) # 2664 taxa (94.04% abundance) 

# associated relative abundances with filters
100*(sum(sample_sums(psdata_RBRA_0.01pct))/sum(sample_sums(psdata_RBRA)))
100*(sum(sample_sums(psdata_RBRA_0.1pct)/sum(sample_sums(psdata_RBRA))))

# choice: to continue downstream analysis with abundance filter that retains ASVs with at least 0.1% of total read abundance
psdata_RBRA_unfiltered <- psdata_RBRA # save unfiltered data
psdata_RBRA <- psdata_RBRA_0.01pct # overwrite psdata_RBRA for abundance filtered data

```

```{r reshape metadata, message=F, echo=T, eval=T, warning=T, include=T, cache=F}

# reformat metadata
metadata = 
  data.frame(sample_data(psdata_RBRA), stringsAsFactors = T) %>% as_tibble() %>% 
  rename_all(. %>% tolower) %>% 
  mutate(combined = factor(paste0(redox_type,"-",sample_type,"-",time))) # will create new combined column with redox_type-sample_type-time information together

meta_formatted = sample_data(metadata)
sample_names(meta_formatted) = metadata$description
sample_data(psdata_RBRA) = meta_formatted

```

### Plotting rarefaction curves

These curves show how well the sequencing effort captured the diversity in the samples.
```{r rarefaction curves, message=F, echo=T, eval=T, warning=T, include=T, cache=F}

## using amp_rarecurve function (does not yield a ggplot object)

# alpha rarefaction curve (https://github.com/MadsAlbertsen/ampvis2/blob/master/R/amp_rarecurve.R)
# source("C:/Users/Rita/Documents/R/Article1/scripts/amp_rarecurve.r")
# amp_rarecurve(psdata_RBRA, color = "sample_type", legend.position = "topleft", xlim = c(0, 50000))
# amp_rarecurve(psdata_RBRA_unfiltered, color = "sample_type", legend.position = "topleft", xlim = c(0, 50000))

#sample_data(psdata_RBRA)$sample <- sample_names(psdata_RBRA)
#df <- as.data.frame(sample_data(psdata_RBRA))
#df <- as_tibble(df)
# df %>% select(sample, everything())
# sample_data(psdata_RBRA) <- sample_data(df)


## using ampvis2 package (yields a ggplot object)

otus_tax = cbind(
  as.data.frame(psdata_RBRA@otu_table),
  as.data.frame(as.matrix(psdata_RBRA@tax_table))) # creates data frame from the otu and taxonomy table phuloseq objects, in which the rows are OTU IDs, the columns are samples, and the last 7 columns are the corresponding taxonomy assigned to the OTUs, named "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species"

met = data.frame(psdata_RBRA@sam_data) # creates data frame from the sample data phyloseq object
met = select(met, sampleid, everything()) # moves sampleid column to the first column

data <- amp_load(otutable = otus_tax,
                 metadata = met)
# Species = ASVs, ideally by the minimum coverage depth full diversity should have been captured captured (all curves reached a plateau)

# minimum coverage of reads per sample
min(sample_sums(psdata_RBRA)) #16905

breaksinoculum <- c("soil", "mun", "ditch", "ind")
valuesfill <- c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")

amp_rarecurve(data = data, 
              color = "sample_type") +
  scale_color_manual (breaks = breaksinoculum,
                      values = valuesfill) +
  theme_classic() +
  theme(axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.width = unit(12, "pt"),
        legend.key.height = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5)) +
  coord_cartesian(xlim = c(0,50000)) +
  geom_vline(xintercept = 16905, size = 0.25)

pdf("C:/Users/Rita/Documents/R/Article1/figures/Figure1_S3.pdf", width = 5, height = 3, useDingbats = F)

amp_rarecurve(data = data, 
              color = "sample_type") +
  scale_color_manual (breaks = breaksinoculum,
                      values = valuesfill) +
  theme_classic() +
  theme(axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.width = unit(12, "pt"),
        legend.key.height = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5)) +
  coord_cartesian(xlim = c(0,50000)) +
  geom_vline(xintercept = 16905, size = 0.25)
```
**Fig x**. Rarefaction curves. Species denotes number of amplicon sequence variants (ASVs) observed from the unrarefied data. Minimum coverage was 16886 reads per sample; a depth at which almost full diversity is captured. Hence, no samples were dropped in later stages.<br>

### Alpha diversity

```{r alpha diversity , message=F, echo=T, eval=T, warning=T, include=T, cache=T}
# Useful info on alpha and beta diversity https://forum.qiime2.org/t/alpha-and-beta-diversity-explanations-and-commands/2282

## all data (0.01% filtered)
psdata_RBRA

## rarefied to 16905 r/s
summary(sample_sums(psdata_RBRA))

set.seed(711)

ps_16905 <- rarefy_even_depth(psdata_RBRA, 16905, rngseed = 711) # lost 0 samples to 100 left

#ps_30000 <- rarefy_even_depth(psdata_RBRA, 30000, rngseed = 711) # lost 6 samples to 30 left

#ps_16886_genus <- tax_glom(ps_16886, "Genus") # 'tax_glom' merges species that have the same taxonomy at a certain taxonomic rank

# calculate richness
metadata <- data.frame(sample_data(ps_16905))
alpha <- estimate_richness(ps_16905, measures = c("Chao1", "Shannon"))
#alpha_genus <- estimate_richness(ps_16886_genus, measures = c("Chao1", "Shannon"))

alpha$sampleid <- sample_names(ps_16905)
alpha_data <- 
  left_join(metadata, alpha, by = "sampleid") %>% 
  rename_all(tolower) 

alpha_summary = alpha_data %>% group_by(combined) %>% summarise(mean_chao = mean(chao1), mean_shannon = mean(shannon), n = n())

## plot alpha populations points

chao1 <- alpha_data %>% 
  ggplot(aes(x=redox_type, y=chao1, color = redox_type)) +
  #geom_boxplot(show.legend = F) + 
  scale_color_manual(values = c("#B2182B", "#D6604D", "#F4A582", "#92C5DE", "#4393C3", "#2166AC")) +
  geom_jitter(position = position_dodge(0.3)) + 
  facet_wrap(~sample_type, nrow = 4) +
  labs(y = "ASV richness (Chao1)", x= "Redox condition") +
  theme_classic() +
  guides(color = guide_legend(nrow=2)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

shannon <- alpha_data %>% 
  ggplot(aes(x=redox_type, y=shannon, color = redox_type)) +
  #geom_boxplot(show.legend = F) + 
  scale_color_manual(values = c("#B2182B", "#D6604D", "#F4A582", "#92C5DE", "#4393C3", "#2166AC")) +
  geom_jitter(position = position_dodge(0.3)) + 
  facet_wrap(~sample_type, nrow = 4) +
  labs(y = "Shannon diversity", x= "Redox condition") +
  theme_classic() +
  guides(color = guide_legend(nrow=2)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) 

library(cowplot)
prow = plot_grid(chao1 + theme(legend.position = "none"), 
                 shannon + theme(legend.position = "none"),
                 align = "vh",
                 labels = c("A", "B"),
                 hjust = -1, 
                 ncol = 2)
# legend_alpha_pop <- get_legend(chao1 + guides(color = guide_legend(ncol = 1)) +
#     theme(legend.position = "right"))
# alpha_popplot <- plot_grid(prow, legend_alpha_pop, ncol=2, rel_widths = c(5, 0.3))
# alpha_popplot
# gridExtra::grid.arrange(chao1, shannon, nrow = 2) 
prow
ggsave(plot = prow, "C:/Users/Rita/Documents/R/Article1/figures/RBRA_proj2_Alpha_diversity_redox_types_rare16886.pdf", width = 8, height = 6)


## plot alpha populations bars

chao1.1 <- alpha_data %>% 
  ggplot(aes(x=redox_type, y=chao1, fill = redox_type)) +
  stat_summary(fun = mean, geom = "bar") +
  scale_fill_manual(values = c("#B2182B", "#D6604D", "#F4A582", "#92C5DE", "#4393C3", "#2166AC")) +
  geom_jitter(position = position_dodge(0.5), pch = 21) + 
  facet_wrap(~sample_type, nrow = 4) +
  labs(y = "ASV richness (Chao1)", x= "Redox condition") +
  theme_classic() +
  guides(color = guide_legend(nrow=2)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
  
shannon.1 <- alpha_data %>% 
  ggplot(aes(x=redox_type, y=shannon, fill = redox_type)) +
  stat_summary(fun = mean, geom = "bar") +
  scale_fill_manual(values = c("#B2182B", "#D6604D", "#F4A582", "#92C5DE", "#4393C3", "#2166AC")) +
  geom_jitter(position = position_dodge(0.5), pch = 21) + 
  facet_wrap(~sample_type, nrow = 4) +
  labs(y = "Shannon diversity", x= "Redox condition") +
  theme_classic() +
  guides(color = guide_legend(nrow=2)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

library(cowplot)
prow.1 = plot_grid(chao1.1 + theme(legend.position = "none"), 
                 shannon.1 + theme(legend.position = "none"),
                 align = "vh",
                 labels = c("A", "B"),
                 hjust = -1, 
                 ncol = 2)
# legend_alpha_pop <- get_legend(chao1 + guides(color = guide_legend(ncol = 1)) +
#     theme(legend.position = "right"))
# alpha_popplot <- plot_grid(prow, legend_alpha_pop, ncol=2, rel_widths = c(5, 0.3))
# alpha_popplot
# gridExtra::grid.arrange(chao1, shannon, nrow = 2) 
prow.1
ggsave(plot = prow.1, "C:/Users/Rita/Documents/R/Article1/figures/RBRA_proj2_Alpha_diversity_redox_types_rare16886_bars.pdf", width = 8, height = 6)

```
**Fig y** Alpha diversity. Bars represent mean values of technical extraction duplicates (dots). Estimates are based on rarefied data at 16886 reads per sample, which captured nearly full diversity (see rarefaction plot)<br>

### Beta diversity

Abundance-based metrics are ignored due to potential large influence of PCR bias. Principal coordinates analysis (PCoA) of Jaccard and unweighted UniFrac distances are presented only, relying on taxon occurrences only. 

```{r total beta diversity, message=F, echo=T, eval=T, warning=T, include=T, cache=F, fig.width=5, fig.height=4}

### Beta diversity analysis first
# input data  = psdata_RBRA

# input is relative abundance table
# relative abundance data
psdata_RBRA_rel <- transform_sample_counts(psdata_RBRA, fun = function(x) x/sum(x))

# transform clr of relative abundances
#psdata_CBUE_clr = transform(psdata_CBUE, transform = "clr")


# ordination
PCoA_BC <- ordinate(psdata_RBRA_rel, method = "PCoA", distance = "bray")
PCoA_Jac <- ordinate(psdata_RBRA_rel, method = "PCoA", distance = "jaccard")
PCoA_uu <- ordinate(psdata_RBRA_rel, method = "PCoA", distance = "uunifrac")
#PCoA_wu <- ordinate(psdata_RBRA_rel, method = "PCoA", distance = "wunifrac")

colnames(sample_data(psdata_RBRA_rel))

## Abundance-based metrics are ignored due to potential large influence of PCR bias.

## Bray_curtis
plot_PCoA_BC <- plot_ordination(physeq = psdata_RBRA_rel, 
                                 ordination = PCoA_BC, 
                  type = "samples", 
                  axes = c(1,2), 
                  color = "redox_type", 
                  shape = "sample_type") +
  scale_color_manual(name = "Redox condition",
                     values = c(brewer.pal(6, "Dark2"),brewer.pal(8,"Paired"))) +
  scale_shape_discrete(name = "Sample type") +
  geom_point(size = 2) +
  ggtitle("Bray-Curtis dissimilarity") +
  theme_classic()

# phylum level
# psdata_RBRA_phylum <- tax_glom(psdata_RBRA, "Phylum")
# psdata_RBRA_phylum_rel <- transform_sample_counts(psdata_RBRA_phylum, fun = function(x) x/sum(x))
# PCoA_Jac_phylum <- ordinate(psdata_RBRA_phylum_rel, method = "PCoA", distance = "jaccard")
# plot_PCoA_Jac_phylum <- plot_ordination(physeq = psdata_RBRA_phylum_rel, 
#                                  ordination = PCoA_Jac_phylum, 
#                   type = "biplot", 
#                   axes = c(1,2), 
#                   color = "Phylum",
#                   shape = "sample_type") +
#   # scale_color_manual(name = NULL,
#   #                    values = c(brewer.pal(6, "Dark2"),brewer.pal(8,"Paired"))) +
#   ggtitle("Jaccard distance") +
#   theme_classic()


plot_PCoA_Jac <- plot_ordination(physeq = psdata_RBRA_rel, 
                                 ordination = PCoA_Jac, 
                  type = "samples", 
                  axes = c(1,2), 
                  color = "redox_type", 
                  shape = "sample_type") +
  scale_color_manual(name = "Redox condition",
                     values = c(brewer.pal(6, "Dark2"),brewer.pal(8,"Paired"))) +
  scale_shape_discrete(name = "Sample type") +
  geom_point(size = 2) +
  ggtitle("Jaccard distance") +
  theme_classic()

plot_PCoA_uu <- plot_ordination(physeq = psdata_RBRA_rel, 
                                ordination = PCoA_uu, 
                                type = "samples", 
                                axes = c(1,2), 
                                color = "redox_type", 
                                shape = "sample_type") + 
  scale_color_manual(name = "Redox condition",
                     values = c(brewer.pal(6, "Dark2"),brewer.pal(8,"Paired"))) +
  scale_shape_discrete(name = "Sample type") +
  geom_point(size = 2) +
  ggtitle("unweighted UniFrac") +
  theme_classic()

# plot_PCoA_wu <- plot_ordination(physeq = psdata_CBUE_rel, 
#                                 ordination = PCoA_wu, 
#                                 type = "samples", 
#                                 axes = c(1,3), 
#                                 color = "biomass_type", 
#                                 shape = "phase") + 
#   scale_color_manual(name = NULL,
#                      values = c(brewer.pal(6, "Dark2"),brewer.pal(8,"Paired"))) +
#   ggtitle("weighted UniFrac") +
#   theme_classic()

#pdf("plot_beta_diversity_OBER_proj1_BODAC.pdf", useDingbats = F, width = 12, height = 7)
prow.2 = plot_grid(plot_PCoA_BC + theme(legend.position = "none"), 
                   plot_PCoA_Jac + theme(legend.position = "none"), 
                   plot_PCoA_uu + theme(legend.position = "none"), 
                   labels  = c("A", "B", "C"), 
                   ncol = 3, 
                   nrow = 1, 
                   align = "hv")

legend_beta <- get_legend(plot_PCoA_Jac + guides(color = guide_legend(ncol = 1), shape = guide_legend(ncol=1)) +
    theme(legend.position = "right"))
beta_plot <- plot_grid(prow.2, legend_beta, ncol=2, rel_widths = c(7.5, 1))

#gridExtra::grid.arrange(plot_PCoA_Jac, plot_PCoA_BC, plot_PCoA_uu, plot_PCoA_wu, nrow=2, ncol=2)
#dev.off()
beta_plot
ggsave(plot = beta_plot, "C:/Users/Rita/Documents/R/Article1/figures/RBRA_proj2_Beta_diversity_redox_types_rare16886_bars.pdf", width = 13, height = 4)

```
**Fig. z**. Microbial community composition based on taxon occurrences. Principal coordinates analysis (PCoA) ordination plots of community composition based on A) Jaccard distance and B) unweighted UniFrac distances. Redox conditions applied and initial origin (sample type) both explain variation in microbial community composition. Aerobic conditions, soil-derived and industrial activated sludge-derived microbial commmunities clustered most consitently. PERMANOVA statistics yet to be performed.<br>

```{r total beta diversity, message=F, echo=T, eval=T, warning=T, include=T, cache=F, fig.width=5, fig.height=4}
# Adonis example



# substract sample metadata from phyloseq object (here psdata_RBRA)
metadata2 <- data.frame(sample_data(psdata_RBRA_rel), stringsAsFactors = T)



# caluclate Jaccard distance matrix
dist_jac = vegdist(t(otu_table(psdata_RBRA_rel)), method = "jaccard", binary = T)

dist_uu = phyloseq::UniFrac(ps_4000, weighted = F)

# test model assumptions (here Redox_type is name of variable of interest; can also be sample_type)
betadisper_jac <- betadisper(dist_jac,metadata2$redox_type)
anova(betadisper_jac)
betadisper_jac <- betadisper(dist_jac,metadata2$sample_type)


# PERMANOVA (permutational multivariate analysis of variance) with the 'adonis' function
#(here Redox_type is name of variable of interest; can also be sample_type, or both separated by a +)



adonis_jac <- adonis(dist_jac ~ redox_type * sample_type, permutations = 999, data = metadata2)
adonis_jac
```

### Relative abundance of Class
```{r rel abund barplot, message=F, echo=T, eval=T, warning=T, include=T, cache=F}

library(RColorBrewer)
colorset <- c("darkblue", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "lightskyblue", "darkgreen",  "khaki2", "firebrick", "brown1", "darkorange1", "cyan1", "royalblue4", "darksalmon", "darkblue", "royalblue4", "dodgerblue3", "steelblue1", "lightskyblue", "darkseagreen", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "brown1", "darkorange1", "cyan1", "darkgrey", "darkblue", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "lightskyblue", "darkgreen", "deeppink", "khaki2", "firebrick", "brown1", "darkorange1", "cyan1", "royalblue4", "darksalmon", "darkblue", "royalblue4", "dodgerblue3", "steelblue1", "lightskyblue", "darkseagreen", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "brown1", "darkorange1", "cyan1", "darkgrey")

#load psmelt function
source("C:/Users/Rita/Documents/R/Article1/scripts/psmelt.r")

#input data
ps_Class <- psmelt(transform_sample_counts(tax_glom(psdata_RBRA, "Class"), fun = function(x) x/sum(x)))

# count n genera n = 129
ps_Class %>% 
  as_tibble() %>% 
  mutate(Class = factor(Class)) %>% 
  pull(Class) %>% 
  levels()


Class_abundances <- ps_Class %>% 
  as_tibble() %>% 
  select(sampleid, redox_type, sample_type, Class, Abundance) %>% 
  mutate(redox_type = as.character(redox_type),
         sample_type = as.character(sample_type)) %>% 
  group_by(sampleid, redox_type, sample_type, Class) %>% 
  summarize(rel_abund = sum(Abundance), .groups = "drop") %>% 
  group_by(redox_type,sample_type, Class) %>% 
  summarize(mean_rel_abund = 100* mean(rel_abund), .groups = "drop") %>% 
  # mutate(Population = factor(Population, 
  #                            levels = c("Aekingerzand", 
  #                                       "Mahazat as-Sayd", 
  #                                       "Taif", 
  #                                       "Kinangop", 
  #                                       "Kedong"))) %>% 
  mutate(Class = str_replace(Class, "(.*)_unClassified", "UnClassified *\\1*"),
         Class = str_replace(Class, "^(\\S*)$", "*\\1*"))


Class_pool <- Class_abundances %>% 
  group_by(Class) %>% 
  summarize(pool = max(mean_rel_abund) < 3, 
            mean = mean(mean_rel_abund), 
            .groups = "drop") 
  
inner_join(Class_abundances, Class_pool, by="Class") %>% 
  mutate(Class = if_else(pool, "Other", Class)) %>% 
  group_by(Class) %>%
  summarize(mean_rel_abund = sum(mean_rel_abund), 
            mean = min(mean),
            .groups = "drop") %>% 
  mutate(Class = factor(Class), 
         Class = fct_reorder(Class, mean, .desc = T))

plot_Class <- inner_join(Class_abundances, Class_pool, by="Class") %>% 
  mutate(Class = if_else(pool, "Other", Class)) %>% 
  group_by(redox_type, sample_type, Class) %>%
  summarize(mean_rel_abund = sum(mean_rel_abund), 
            mean = min(mean),
            .groups = "drop") %>% 
  mutate(Class = factor(Class), 
         Class = fct_reorder(Class, mean, .desc = T),
         #Class = fct_shift(Class, n=1),
         redox_type = factor(redox_type),
         redox_type = fct_relevel(redox_type, rev(c("inoculum","aerobic", "nitrate", "iron", "sulphate", "methanogenic")))) %>% 
ggplot(aes(x=redox_type, y = mean_rel_abund, fill = Class)) +
  scale_fill_manual(values = colorset) +
  geom_col() +
  scale_fill_manual(name=NULL, 
                    values = c(brewer.pal(6,"Dark2"),brewer.pal(8,"Paired"), "grey50", colorset[c(1:10)])) + # with palette
                    #values = colorset) +
  scale_y_continuous(expand = c(0,0)) +
  labs(x=NULL, y="Mean Relative Abundance (%)") +
  # scale_x_discrete(breaks = c("Aekingerzand", 
  #                             "Mahazat as-Sayd", 
  #                             "Taif", 
  #                             "Kinangop", 
  #                             "Kedong"),
  #                  labels = c("Aekingerzand<br>*The Netherlands*", 
  #                             "Mahazat as-Sayd<br>*Saudi Arabia*", 
  #                             "Taif<br>*Saudi Arabia*", 
  #                             "Kinangop<br>*Kenya*", 
  #                             "Kedong<br>*Kenya*")) +
  theme_classic() +
  theme(axis.text.x = element_markdown(),
        legend.text = element_markdown(),
        legend.key.size = unit(10, "pt")) +
  facet_wrap(~sample_type, ncol = 4) +
  coord_flip()

 
prow.3 = plot_grid(plot_Class + theme(legend.position = "none"), 
                   ncol = 1, 
                   nrow = 1, 
                   align = "hv")

legend_bar <- get_legend(plot_Class + guides(fill = guide_legend(ncol = 1)) +
    theme(legend.position = "right"))
bar_plot <- plot_grid(prow.3, legend_bar, ncol=3, rel_widths = c(8, 2,0.3))

#gridExtra::grid.arrange(plot_PCoA_Jac, plot_PCoA_BC, plot_PCoA_uu, plot_PCoA_wu, nrow=2, ncol=2)
#dev.off()
bar_plot
ggsave(plot = bar_plot, "C:/Users/Rita/Documents/R/Article1/figures/RBRA_proj2_Stacked_bar_Rel_abund_redox_types_rare16886_bars.pdf", width = 12, height = 4)
```

### TODO
Check validity of removed contaminants (i.e. psdata_contam).

