---
title: "Influence of redox conditions and inoculum source on the micropollutants biodegradation by soil and activated sludge communities"
author: "Pieter van Veelen"
date: "7/27/2021"
output: html_document
---

This is an R Markdown document contains an analysis of microbial community dynamics of an experiment of Rita H.R. Branco in collaboration with H. Pieter J. van Veelen.

## Install packages

```{r setup, include=T}
knitr::opts_chunk$set(echo = TRUE)

options(scipen = 999)

setwd("C:/Users/Rita/Documents/R/Article1/scripts")

# # packages required
# packages <- c("phyloseq", "qiime2R", "tidyverse", "magrittr", "devtools", "qiime2R", "here", "breakaway", "DivNet", "openxlsx", "ape", "vegan", "ggtext", "cowplot", "RColorBrewer", "microbiome", "lme4", "lmerTest","decontam", "remotes", "ampvis2", "speedyseq", "ggh4x", "kableExtra", "phia", "MASS", "ggord", "glue", "lubridate")
# 
# # install packages not yet installed
# installed_packages <- packages %in% rownames(installed.packages())
# if (any(installed_packages == FALSE)) {
#   install.packages(packages[!installed_packages])
# }
# 
# # packages not available for R 4.1.2
# ## phyloseq
# if(!requireNamespace("BiocManager")){install.packages("BiocManager")}
# BiocManager::install("phyloseq")
# 
# ## qiime2R
# if (!requireNamespace("devtools", quietly = TRUE)){install.packages("devtools")}
# devtools::install_github("jbisanz/qiime2R")
# 
# ## breakaway
# install.packages("devtools")
# devtools::install_github("adw96/breakaway")
# 
# ## DivNet
# remotes::install_github("adw96/breakaway")
# remotes::install_github("adw96/DivNet")
# 
# ## microbiome
# library(BiocManager)
# BiocManager::install("microbiome")
# 
# ## decontam
# if (!requireNamespace("BiocManager", quietly = TRUE))
# install.packages("BiocManager")
# BiocManager::install("decontam")
# 
# ## ampvis2
# if (!requireNamespace("ampvis2", quietly = TRUE))
# install.packages("ampvis2")
# remotes::install_github("MadsAlbertsen/ampvis2", Ncpus = 6)
# 
# ## speedyseq
# if (!requireNamespace("speedyseq", quietly = TRUE))
#     install.packages("speedyseq")
# remotes::install_github("mikemc/speedyseq")
# 
# ## ggord
# if (!requireNamespace("ggord", quietly = TRUE))
# install.packages("remotes")
# remotes::install_github("fawda123/ggord")
```

## Exploratory data analysis

### Loading libraries

```{r library loading, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# load required packages
library(phyloseq)
library(qiime2R)
library(tidyverse)
library(magrittr)
library(devtools)
library(qiime2R)
library(here)
library(breakaway)
library(DivNet)
library(openxlsx)
library(ape)
library(vegan)
library(ggtext)
library(cowplot)
library(RColorBrewer)
library(microbiome)
library(lme4)
library(lmerTest)
library(decontam)
library(remotes)
library(ampvis2)
library(speedyseq)
library(ggh4x)
library(ggord)
library(glue)
library(lubridate)
library(phia)
library(kableExtra)
library(MASS)
```

### Data import

Below, the input data are imported in R. These files are output files from our QIIME2 (v.2019.10) sequence data pre-processing workflow. Scripts and parameter settings are found in separate bash files.

```{r import data, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# creating phyloseq objects with 
physeq = qza_to_phyloseq(features = "C:/Users/Rita/Documents/R/Article1/input_data/RBRA_proj2_Q13921_16S_515F926R_20210701_table.qza",
                        tree = "C:/Users/Rita/Documents/R/Article1/input_data/RBRA_proj2_Q13921_16S_515F926R_20210701_rooted-tree.qza",
                        taxonomy = "C:/Users/Rita/Documents/R/Article1/input_data/RBRA_proj2_Q13921_16S_515F926R_20210701_taxonomy_NB_classifier_SILVA_132_99_16S_515F-926R_QIIME2-2019.10.qza",
                        metadata = "C:/Users/Rita/Documents/R/Article1/input_data/RBRA_proj2_Q13921_16S_515F926R_20210701@metadata_formatted_v2.txt")
```

### Cleaning data set

The following steps are subsequently performed to clean the data: 1) bifurcating tree using ape package; 2) cleaning up the metadata; 3) replacing taxonomic strings that are empty, NA, metagenome, ambiguous taxa; 4) decontaminating based on frequency and prevalence using the DECONTAM package; 5) split blanks from samples; 

#### 1) Resolve phylogenetic tree

Number of tips = number internal nodes + 1. If this does not happens, then nodes from the phylogenetic tree have to be splited/bifurcated.

```{r clean phylogeny, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# evaluate tree topology
is.binary(phy_tree(physeq))

# if FALSE -> resolve polychotomous nodes
phy_tree_resolved <- multi2di(phy_tree(physeq))
is.binary(phy_tree_resolved)

# create new phy_tree
tree2 <- phy_tree_resolved

# merge new phy_tree object with sample_data and otu_table into new phyloseq object
psdata_RBRA <- merge_phyloseq(otu_table(physeq), sample_data(physeq), tax_table(physeq), tree2)
```

#### 2) Clean-up metadata

```{r clean metadata, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
sample_data(psdata_RBRA)$Redox_type <- factor(sample_data(psdata_RBRA)$Redox_type,levels=c("inoculum", "aerobic", "nitrate", "iron", "sulphate", "methanogenic"))

sample_data(psdata_RBRA)$Sample_type <- factor(sample_data(psdata_RBRA)$Sample_type, levels=c("soil", "mun", "ditch", "ind", "blank"))

names(sample_data(psdata_RBRA)) <- tolower(names(sample_data(psdata_RBRA)))
```

#### 3) Clean-up taxa data

```{r clean taxanomy, message=F, echo=T, eval=T, warning=F, include=T, cache=F}
# remove Chloroplast and Mitochondia
physeq1 <- subset_taxa(psdata_RBRA, Order != "Chloroplast")
physeq2 <- subset_taxa(physeq1, Family != "Mitochondria")

# clean taxonomy tags with no information, specify NA taxon name tags to last known taxon names
tax.clean <- data.frame(tax_table(physeq2))

tax.clean2 <- tax.clean
tax.clean2$Kingdom <- as.character(tax.clean2$Kingdom)
tax.clean2$Phylum <- as.character(tax.clean2$Phylum)
tax.clean2$Class <- as.character(tax.clean2$Class)
tax.clean2$Order <- as.character(tax.clean2$Order)
tax.clean2$Family <- as.character(tax.clean2$Family)
tax.clean2$Genus <- as.character(tax.clean2$Genus)
tax.clean2$Species <- as.character(tax.clean2$Species)

tax.clean = tax.clean2 %>% mutate_all(funs(str_replace(., "Ambiguous_taxa", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "metagenome", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "uncultured", "")))
tax.clean = tax.clean %>% mutate_all(funs(str_replace(., "unidentified", "")))
tax.clean[is.na(tax.clean)] <- ""

# function replace name tags from [https://github.com/joey711/phyloseq/issues/850]
for (i in 1:nrow(tax.clean)){
  if (tax.clean[i,2] == ""){
    Kingdom <- paste("Kingdom_", tax.clean[i,1], sep = "")
    tax.clean[i, 2:7] <- Kingdom
  } else if (tax.clean[i,3] == ""){
    Phylum <- paste("Phylum_", tax.clean[i,2], sep = "")
    tax.clean[i, 3:7] <- Phylum
  } else if (tax.clean[i,4] == ""){
    Class <- paste("Class_", tax.clean[i,3], sep = "")
    tax.clean[i, 4:7] <- Class
  } else if (tax.clean[i,5] == ""){
    Order <- paste("Order_", tax.clean[i,4], sep = "")
    tax.clean[i, 5:7] <- Order
  } else if (tax.clean[i,6] == ""){
    Family <- paste("Family_", tax.clean[i,5], sep = "")
    tax.clean[i, 6:7] <- Family
  } else if (tax.clean[i,7] == ""){
    tax.clean$Species[i] <- paste("Genus_",tax.clean$Genus[i], sep = "_")
  }
}

psdata_newTaxtab <- psdata_RBRA
tax_table(psdata_newTaxtab) <- as.matrix(tax.clean)

# put cleaned tax_table into phyloseq object
tax_table(physeq2) <- as.matrix(tax.clean)

psdata_RBRA <- physeq2
```

#### 4) Remove contaminants with DECONTAM

```{r decontam, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# add sample_or_control column
sample_data(psdata_RBRA)$is.neg <- sample_data(psdata_RBRA)$sample_type == "blank"
sample_data(psdata_RBRA)$sample_or_control <- factor(if_else(sample_data(psdata_RBRA)$sample_type == "blank", "control", "sample"))

# replace DNA concentration = 0 for a very low biomass samples -> low DNA amount
sample_data(psdata_RBRA)$dna_ng_ml <- if_else(sample_data(psdata_RBRA)$dna_ng_ml == 0, sample_data(psdata_RBRA)$dna_ng_ml+0.0001, sample_data(psdata_RBRA)$dna_ng_ml)

# contaminants based on prevalence due to low DNA concentration
contamdf.prev <- isContaminant(psdata_RBRA, method="prevalence", neg="is.neg", threshold=0.1)
n_contam_prev <- table(contamdf.prev$contaminant) # 2 contaminants based on prevalence

# saving contaminant and decontaminated data
psdata_contam <-  prune_taxa(contamdf.prev$contaminant, psdata_RBRA)
psdata_decontam <- prune_taxa(!contamdf.prev$contaminant, psdata_RBRA)
```

#### 5) Remove blanks from the dataset

```{r select samples, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# add SampleID
sample_data(psdata_decontam)$sampleid <- sample_names(psdata_decontam)

# keep the taxa that are present on one ore more of the samples
psdata_RBRA_only <- 
  prune_taxa(
    taxa_sums(
      subset_samples(psdata_decontam, sample_type != "blank")) > 0, 
      subset_samples(psdata_decontam, sample_type != "blank"))

psdata_RBRA <- prune_taxa(taxa_sums(psdata_RBRA_only) >0, psdata_RBRA_only)
```

### Rarefaction curves

#### Filter on abundance 

```{r filter on abundance, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# relative abundance data
psdata_RBRA_rel <- transform_sample_counts(psdata_RBRA, fun = function(x) x/sum(x)) # 10177 taxa

# continue downstream analysis with abundance filter that retains ASVs with at least 0.1% of total read abundance -> keeps above 99% abundance when removing rare data
psdata_RBRA_0.01pct <- prune_taxa(taxa_sums(psdata_RBRA_rel) > 0.0001, psdata_RBRA) # 7590 taxa (99.46% abundance)

psdata_RBRA <- psdata_RBRA_0.01pct
```

#### Reshape metadata

```{r reshape metadata, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
metadata = 
  data.frame(sample_data(psdata_RBRA), stringsAsFactors = T) %>% as_tibble() %>% 
  rename_all(. %>% tolower) %>% 
  mutate(combined = factor(paste0(redox_type,"-",sample_type,"-",time)))

meta_formatted = sample_data(metadata)
sample_names(meta_formatted) = metadata$description
sample_data(psdata_RBRA) = meta_formatted
```

#### Plotting rarefaction curves

```{r rarefaction curves, message=F, echo=T, eval=T, warning=F, include=T, cache=F, fig.width=6, fig.height=4}
# data frame in which the rows are OTU IDs, the columns are samples, and the last 7 columns are the corresponding taxonomy assigned to the OTUs
otus_tax = cbind(
  as.data.frame(psdata_RBRA@otu_table),
  as.data.frame(as.matrix(psdata_RBRA@tax_table)))

# data frame with metadata
met = data.frame(psdata_RBRA@sam_data)
met = dplyr::select(met, sampleid, everything())

# load data
data <- amp_load(otutable = otus_tax,
                 metadata = met)

# minimum coverage of reads per sample
min_cover <- min(sample_sums(psdata_RBRA)) #16905

# rarefaction curves plot
breaksinoculum <- c("soil", "mun", "ditch", "ind")
valuesfill <- c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")

amp_rarecurve(data = data, 
              color = "sample_type") +
  scale_color_manual (breaks = breaksinoculum,
                      values = valuesfill) +
  theme_classic() +
  theme(axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5)) +
  coord_cartesian(xlim = c(0,50000)) +
  geom_vline(xintercept = 16905, size = 0.25)
```

**Figure S3**. Rarefaction curves. Minimum coverage was 16905 reads per sample; a depth at which almost full diversity is captured. Hence, no samples were dropped in later stages.<br>

### Alpha diversity

Abundance-based metrics are ignored due to potential large influence of PCR bias. ASV richness (Chao1) is presented only, relying on taxon occurrences only.

```{r alpha diversity , message=F, echo=T, eval=T, warning=T, include=T, cache=T, fig.width=4, fig.height=6}
# data rarefied to 16905 r/s
set.seed(711)

ps_16905 <- rarefy_even_depth(psdata_RBRA, 16905, rngseed = 711)

# calculate richness
metadata <- data.frame(sample_data(ps_16905))
alpha <- estimate_richness(ps_16905, measures = c("Chao1"))

alpha$sampleid <- sample_names(ps_16905)
alpha_data <- 
  left_join(metadata, alpha, by = "sampleid") %>% 
  rename_all(tolower) 

alpha_summary = alpha_data %>% group_by(combined) %>% summarise(mean_chao = mean(chao1), n = n())

# plot alpha populations bars
alpha_data %>% 
  ggplot(aes(x=redox_type, y=chao1)) +
  stat_summary(fun = mean, geom = "bar", mapping = aes(fill = sample_type)) +
  geom_point(mapping = aes(shape = redox_type)) +
  scale_fill_manual(values = c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")) +
  scale_shape_manual(values = c(17,0,1,4,5,6)) +
  facet_wrap(~sample_type, nrow = 4) +
  labs(y = "ASV richness (Chao1)", x= "Redox condition") +
  theme_classic() +
  theme(axis.text = element_text(size = 7, colour = "black"),
        axis.ticks.x = element_line(size = 0.35),
        axis.ticks.y = element_line(size = 0.35),
        axis.text.x	= element_text(angle = 45, hjust = 1),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.position = "none",
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))
```

**Figure 4A**. Microbial community diversity across inocula and inocula-derived communities under different redox conditions. A) Alpha diversity expressed as estimated ASV richness (Chao1). Bars represent mean values of technical extraction duplicates.<br>

### Beta diversity

Abundance-based metrics are ignored due to potential large influence of PCR bias. Principal coordinates analysis (PCoA) of Jaccard and unweighted UniFrac distances are presented only, relying on taxon occurrences only. Due to the low number of replicates no PERMANOVA was performed.

```{r total beta diversity, message=F, echo=T, eval=T, warning=T, include=T, cache=F, fig.width=10, fig.height=5}
# relative abundance data
psdata_RBRA_rel <- transform_sample_counts(psdata_RBRA, fun = function(x) x/sum(x))

# ordination
PCoA_Jac <- ordinate(psdata_RBRA_rel, method = "PCoA", distance = "jaccard")
PCoA_uu <- ordinate(psdata_RBRA_rel, method = "PCoA", distance = "uunifrac")

# plot PCoA Jaccard distances
plot_PCoA_Jac <- plot_ordination(physeq = psdata_RBRA_rel, 
                                 ordination = PCoA_Jac, 
                                 type = "samples", 
                                 axes = c(1,2), 
                                 shape = "redox_type",
                                 color = "sample_type") +
  scale_color_manual(name = "Microbial culture",
                     values = c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")) +
  scale_shape_manual(name = "Redox condition",
                       values = c(17,0,1,4,5,6)) +
  geom_point(size = 2) +
  ggtitle("Jaccard distance") +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))

# plot PCoA unweighted UniFrac distances
plot_PCoA_uu <- plot_ordination(physeq = psdata_RBRA_rel, 
                                ordination = PCoA_uu, 
                                type = "samples", 
                                axes = c(1,2), 
                                shape = "redox_type",
                                color = "sample_type") + 
  scale_color_manual(name = "Microbial culture",
                     values = c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")) +
  scale_shape_manual(name = "Redox condition",
                       values = c(17,0,1,4,5,6)) +
  geom_point(size = 2) +
  ggtitle("unweighted UniFrac") +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))

# plot PCoA Jaccard and unweighted UniFrac distances
PCoA = plot_grid(plot_PCoA_Jac + theme(legend.position = "none"),
                   plot_PCoA_uu + theme(legend.position = "none"),
                   labels  = c("B", "C"),
                   ncol = 2,
                   nrow = 1,
                   align = "hv")

legend_beta <- get_legend(plot_PCoA_Jac)

plot_grid(PCoA, legend_beta, ncol=2, rel_widths = c(5, 1))
```

**Figure 4B and 4C**. Microbial community diversity across inocula and inocula-derived communities under different redox conditions. Principal coordinates analysis (PCoA) ordination plots of community composition based on B) Jaccard and C) unweighted UniFrac distances, respectively.<br>

#### Distances between inoculum and redox derived community

```{r ditances inoculum vs redox, message=F, echo=T, eval=T, warning=F, include=T, cache=F, fig.width=10, fig.height=5}
# subtract sample metadata from phyloseq object
metadata2 <- data.frame(sample_data(psdata_RBRA_rel), stringsAsFactors = T)

# calculate distance matrices
dist_jac = vegdist(t(otu_table(psdata_RBRA_rel)), method = "jaccard", binary = T)

dist_uu = phyloseq::UniFrac(psdata_RBRA_rel, weighted = F)

# reshape to long format
jaccard_long =
dist_jac %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_a") %>%
  pivot_longer(-sample_a, names_to = "sample_b", values_to = "distance") %>%
  mutate(metric = "jaccard")

uu_long =
dist_uu %>%
  as.matrix() %>%
  as_tibble(rownames = "sample_a") %>%
  pivot_longer(-sample_a, names_to = "sample_b", values_to = "distance") %>%
  mutate(metric = "uu")

# define consecutive sampling moments
comp_inoculum = c("inoculum_to_aerobic", "inoculum_to_nitrate", "inoculum_to_iron", "inoculum_to_sulphate", "inoculum_to_methanogenic")

# reshape distance data and combine with metadata
all_dists =
bind_rows(uu_long, jaccard_long) %>%
  filter(sample_a != sample_b) %>% # remove self-comparisons
  inner_join(., metadata2 %>% dplyr::select(sampleid, redox_type, sample_type), by = c("sample_a" = "sampleid")) %>%
  rename(redox_type_a = redox_type,
         sample_type_a = sample_type) %>% 
   inner_join(., metadata2 %>% dplyr::select(sampleid, redox_type, sample_type), by = c("sample_b" = "sampleid")) %>%
  rename(redox_type_b = redox_type,
         sample_type_b = sample_type) %>% 
  mutate(comp = paste0(redox_type_a, "_to_", redox_type_b)) %>%
  filter(comp %in% comp_inoculum) %>% 
  mutate(comp = factor(comp, levels = comp_inoculum)) %>%
  filter(sample_type_a == sample_type_b)

# plot Jaccard distances between inoculum and redox derived community
comp_inoc_jac <- all_dists %>%
  filter(metric == "jaccard") %>% 
  ggplot(aes(x = comp, y = distance, color = sample_type_a)) +
  geom_point(size = 2, alpha = 0.6, position = position_dodge(0.05)) +
  ggtitle("Jaccard distances") +
  scale_color_manual(values = c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")) +
  scale_fill_manual(values = c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")) +
  scale_y_continuous(limits = c(0.3, 0.9), 
                     breaks = seq(0.3, 0.9, 0.2)) +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.text.x = element_text(angle = 45, hjust = 1),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(5, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5),
        legend.position = "right") +
  labs(y = "Distance",
       x = NULL)

# plot unweighted UniFrac distances between inoculum and redox derived community
comp_inoc_uu <- all_dists %>%
  filter(metric == "uu") %>% 
  ggplot(aes(x = comp, y = distance, color = sample_type_a)) +
  geom_point(size = 2, alpha = 0.6, position = position_dodge(0.05)) +
  ggtitle("unweighted UniFrac distances") +
  scale_color_manual(values = c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")) +
  scale_fill_manual(values = c("#CA0020", "#F4A582", "#92C5DE", "#0571B0")) +
  scale_y_continuous(limits = c(0.3, 0.9), 
                     breaks = seq(0.3, 0.9, 0.2)) +
  theme_classic() +
  theme(plot.title = element_text(size = 10),
        axis.text = element_text(size = 7, colour = "black"),
        axis.text.x = element_text(angle = 45, hjust = 1),
        axis.ticks = element_line(size = 0.25),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text	= element_text(size = 7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(5, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5),
        legend.position = "right") +
  labs(y = "Distance",
       x = NULL)

# plot Jaccard and unweighted UniFrac distances between inoculum and redox derived community
comp = plot_grid(comp_inoc_jac + theme(legend.position = "none"),
                   comp_inoc_uu + theme(legend.position = "none"),
                   labels  = c("E", "F"),
                   ncol = 2,
                   nrow = 1,
                   align = "hv")

legend_comp <- get_legend(comp_inoc_jac)

plot_grid(comp, legend_comp, ncol=2, rel_widths = c(5, 1))
```

**Figure 4D and 4E**. D) Jaccard and E) unweighted UniFrac distances between each inoculum and respective derived microbial communities under different redox conditions, this difference between D) and E) represents the phylogenetic clustering of the observed ASVs.<br>

### Capscale analysis

Capscale was performed with partialling out sample type, i.e. focused on redox condition

```{r partial analysis, message=F, echo=T, eval=T, warning=T, include=F, cache=F}
# load phyloseq object
phy_rita = psdata_RBRA

# chemical data
chem_data = readxl::read_xlsx("C:/Users/Rita/Documents/R/Article1/input_data/Article1_chemical_data.xlsx")

# sample_data
sam_dat = 
  sample_data(phy_rita) %>% data.frame() %>% dplyr::select(SampleID = sampleid, everything()) %>% tibble

# joined data
chem_data_added = sam_dat %>% inner_join(., chem_data, by = "SampleID")

# filter out inocula because they do not have data on micropollutant removal
is_innoculum = setdiff(sam_dat$SampleID, chem_data$SampleID)

# overwriting old metadata
phy_rita_redox = subset_samples(phy_rita, !sample_names(phy_rita) %in% is_innoculum)
phy_rita_redox = prune_taxa(taxa_sums(phy_rita_redox)>0, phy_rita_redox)

sam_data = sample_data(chem_data_added)
sample_names(sam_data) = chem_data_added$SampleID
sample_data(phy_rita_redox) = sam_data

# rarefy data
min_ss = min(sample_sums(phy_rita_redox))

set.seed(711)
phy_rita_redox_16095 = rarefy_even_depth(phy_rita_redox, sample.size = min_ss, rngseed = 711, trimOTUs = T)

# total number of reads
tot_read = phy_rita_redox_16095 %>% sample_sums(.) %>% sum()

# create taxa acronyms
df_taxa_names = data.frame(OTU = taxa_names(phy_rita_redox_16095),
                           ASV = paste0("ASV_",seq(1:ntaxa(phy_rita_redox_16095))))

# rename colnames with acronyms
abund = t(otu_table(phy_rita_redox_16095))
cap_data = abund/rowSums(abund)
colnames(cap_data) = df_taxa_names$ASV

rownames(chem_data_added) = chem_data_added$SampleID

# predictors
chem_pred = 
  chem_data_added %>% 
  dplyr::select(Sample_type, Redox_type, GAB:MET) %>% 
  mutate(MET_ESA = as.numeric(MET_ESA),
         MET_OA = as.numeric(MET_OA)) %>% 
  mutate(across(where(is.numeric), ~if_else(.<0, 0, .))) %>% 
  mutate(across(where(is.numeric), ~scale(.)),
         Redox_type = factor(Redox_type, levels = c("aerobic", "nitrate", "iron", "sulphate", "methanogenic")))

# model definitions (all predictor removal efficiencies are added)
redox.model <- capscale(cap_data ~ 
                         ACK + CBZ + 
                          MCPP + `24_D` + ANP +
                          BTZ + `1H_BTR`+ CLZ_DP + CLZ_MDP +
                          CLZ + MET_ESA +
                          MET_OA + MET +
                          Condition(Sample_type), # partial analysis now
                        chem_pred[,-2], 
                        dist="jaccard",
                        na.action = na.exclude)

vif_cca <- vif.cca(redox.model)
```

```{r model selection, message=F, echo=T, eval=T, warning=T, include=F, cache=F}
# model selection
selectedMod_red <- step(redox.model)
smry = summary(redox.model)
```

```{r ANOVA table, message=F, echo=T, eval=T, warning=T, include=T, cache=F}
# general ANOVA
anova_tab_gen = anova(selectedMod_red) %>% as_tibble(rownames = "terms") 

# redox ANOVA
anova_tab_red = anova(selectedMod_red, by = "terms") %>% as_tibble(rownames = "terms")
anova_tab_red = anova_tab_red %>% 
  mutate(R2 = 100*(SumOfSqs/sum(SumOfSqs))) %>% 
  mutate(fdr_q = p.adjust(`Pr(>F)`, "fdr")) %>% 
  filter(fdr_q < 0.1 | is.na(fdr_q)) # correct p values for multiple testing filter(fdr_q < 0.1)

anova_tab_red
```
**Table S5**.Constrained ordination using db-RDA (Jaccard distances). Associations between microbial community membership and MP removal efficiencies. Terms = MPs with FDR q<0.1 (stepwise selection).<br>

```{r capscale analysis - CAP plot, message=F, echo=T, eval=T, warning=T, include=T, cache=F, fig.width=12, fig.height=6}
# plot dbRDA model focused on redox_conditions
plot_data = chem_data_added %>% dplyr::select(SampleID, Sample_type,Redox_type, any_of(smry$biplot %>% rownames()))
df1  <- data.frame(smry$sites[,1:2]) %>% as_tibble(rownames = "SampleID") # CAP1 and CAP2
plot_data = df1 %>% inner_join(., plot_data, by = "SampleID" )

## plot CAP ordinations
### axis 1 vs 2
ord_red_ax12 = 
  ggord(redox.model,
        xlims = c(-1.1, 0.5),
        ylims = c(-0.8, 0.6),
        arrow = 0.2,
        axes = c("1","2"),
        grp_in = plot_data$Redox_type,
        ellipse =F, 
        coord_fix = T,
        addsize = 0,
        add_col = NULL,
        ext = 1.2,
        vec_ext = 0.8,
        var_sub = anova_tab_red$terms[-length(anova_tab_red$terms)]) +
  labs(title = "Partial dbRDA - effect of Redox condition; CAP1 vs CAP2")

### axis 1 vs 3
ord_red_ax13 = 
  ggord(redox.model,
        xlims = c(-1.1, 0.5),
        ylims = c(-0.9, 0.6),
        arrow = 0.2,
        axes = c("1","3"),
        grp_in = plot_data$Redox_type,
        ellipse =F, 
        coord_fix = T,
        addsize = 0,
        ext = 1.2,
        vec_ext = 0.8,
        var_sub = anova_tab_red$terms[-length(anova_tab_red$terms)]) +
  labs(title = "Partial dbRDA - effect of Redox condition; CAP1 vs CAP3")

### make ggord, remove points layer and add shapes etc
ord_red_ax12$layers[[1]] <- NULL # to adjust shape as well
ord_red_ax13$layers[[1]] <- NULL # to adjust shape as well

ord_red_ax12$data <- ord_red_ax12$data %>% 
  mutate(newgrp = plot_data$Sample_type, # add extra group for sample_type
         samplecheck = plot_data$SampleID,
         Groups = factor(Groups, levels = c("aerobic", "nitrate", "iron", "sulphate", "methanogenic")),
         newgrp = factor(newgrp, levels = c("soil", "mun", "ditch", "ind")))

ord_red_ax13$data <- ord_red_ax13$data %>% 
  mutate(newgrp = plot_data$Sample_type, # add extra group for sample_type
         samplecheck = plot_data$SampleID,
         Groups = factor(Groups, levels = c("aerobic", "nitrate", "iron", "sulphate", "methanogenic")),
         newgrp = factor(newgrp, levels = c("soil", "mun", "ditch", "ind")))

# plot CAP ordinations again
shapes_red = c(22, 21, 4, 24, 25)
colors_red = c("#C92127", "#F3A484", "#93C4DD", "#0772B2")

plot_ord_red_ax12 = 
  ord_red_ax12 + geom_point(aes(shape = Groups, colour = newgrp, group = newgrp), size=2, stroke=1.5, alpha=1) + 
  scale_shape_manual(name = "",
                     values = shapes_red) +
  scale_color_manual(name = "",
                     values = colors_red,
                     labels = c("Soil","Mun AS","Ditch", "Ind AS")) + 
  guides(fill = guide_legend(override.aes= list(shape = NA))) + 
  theme(legend.title = element_blank())

plot_ord_red_ax13 = 
  ord_red_ax13 + geom_point(aes(shape = Groups, colour = newgrp, group = newgrp), size=2, stroke=1.5, alpha=1) + 
  scale_shape_manual(name = "",
                     values = shapes_red) +
  scale_color_manual(name = "",
                     values = colors_red,
                     labels = c("Soil","Mun AS","Ditch", "Ind AS")) + 
  guides(fill = guide_legend(override.aes= list(shape = NA))) + 
  theme(legend.title = element_blank())

legend_bar_CAP = 
  get_legend(plot_ord_red_ax12 + theme( legend.position = "bottom",
                                        legend.box = "vertical",
                                        legend.margin=margin(),
                                        legend.key.size = unit(1, 'cm'),
                                        legend.key.height = unit(1, 'cm'),
                                        legend.key.width = unit(1, 'cm'),
                                        legend.title = element_text(size=0),
                                        legend.text = element_text(size=12)))

CAP_plot = 
  plot_grid(
    plot_ord_red_ax12 + theme(legend.position = "none"), 
    plot_ord_red_ax13 + theme(legend.position = "none"), 
    labels  = c("A", "B"), 
    ncol = 2, 
    nrow = 1, 
    align = "hv")


plot_grid(CAP_plot, legend_bar_CAP, ncol=1, nrow = 2, rel_heights = c(5, 1))
```

**Figure 5**. Redundancy analysis linking micropollutant removal efficiency to microbial community composition. A) CAP1 vs CAP2 and B) CAP1 vs CAP3.<br>

### Relative abundance

```{r rel abund barplot class, message=F, echo=T, eval=T, warning=T, include=T, cache=F, fig.width=8, fig.height=4}
# input data
ps_Class <- psmelt(transform_sample_counts(tax_glom(psdata_RBRA, "Class"), fun = function(x) x/sum(x)))

# relative abundance based on Class
class_count <- ps_Class %>% 
  as_tibble() %>% 
  mutate(Class = factor(Class)) %>% 
  pull(Class) %>% 
  levels() # count class n = 123

Class_abundances <- ps_Class %>% 
  as_tibble() %>% 
  dplyr::select(sampleid, redox_type, sample_type, Class, Abundance) %>% 
  mutate(redox_type = as.character(redox_type),
         sample_type = as.character(sample_type)) %>% 
  group_by(sampleid, redox_type, sample_type, Class) %>% 
  summarize(rel_abund = sum(Abundance), .groups = "drop") %>% 
  group_by(redox_type,sample_type, Class) %>% 
  summarize(mean_rel_abund = 100* mean(rel_abund), .groups = "drop") %>% 
  mutate(Class = str_replace(Class, "(.*)_unClassified", "UnClassified *\\1*"),
         Class = str_replace(Class, "^(\\S*)$", "*\\1*"))

Class_pool <- Class_abundances %>% 
  group_by(Class) %>% 
  summarize(pool = max(mean_rel_abund) < 3, 
            mean = mean(mean_rel_abund), 
            .groups = "drop")

# plot relative abundance based on Class
plot_Class <- inner_join(Class_abundances, Class_pool, by="Class") %>% 
  mutate(Class = if_else(pool, "Other", Class)) %>% 
  group_by(redox_type, sample_type, Class) %>%
  summarize(mean_rel_abund = sum(mean_rel_abund), 
            mean = min(mean),
            .groups = "drop") %>% 
  mutate(Class = factor(Class), 
         Class = fct_reorder(Class, mean, .desc = T),
         redox_type = factor(redox_type),
         redox_type = fct_relevel(redox_type, rev(c("inoculum","aerobic", "nitrate", "iron", "sulphate", "methanogenic"))),
         sample_type = factor(sample_type),
         sample_type = fct_relevel(sample_type, rev(c("ind", "ditch", "mun", "soil")))) %>% 
ggplot(aes(x=redox_type, y = mean_rel_abund, fill = Class)) +
  geom_col() +
  scale_fill_manual(name=NULL, 
                    values = c("#B2182B", "#D6604D", "#2166AC", "#4393C3", "#4D4D4D", "#878787", "#01665E", "#35978F", "#ABDDA4", "#E6F598", "#8C510A", "#BF812D", "#C51B7D", "#DE77AE", "#4D9221", "#7FBC41",  "#762A83", "#9970AB", "#FDAE61", "#FEE090",  "#1B7837", "#5AAE61")) +
  facet_wrap(~sample_type, ncol = 4) +
  scale_y_continuous(expand = c(0,0)) +
  labs(x="Redox condition", y="Mean Relative Abundance (%)") +
  theme_classic() +
  theme(axis.text = element_text(size = 7, colour = "black"),
        axis.ticks.x = element_line(size = 0.35),
        axis.ticks.y = element_line(size = 0.35),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text = element_markdown(size=7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5)) +
  coord_flip()

class = plot_grid(plot_Class + theme(legend.position = "none"), 
                   ncol = 1, 
                   nrow = 1, 
                   align = "hv")

legend_class <- get_legend(plot_Class + guides(fill = guide_legend(ncol = 8)) +
    theme(legend.position = "right"))

plot_grid(class, legend_class, nrow=2, rel_heights = c(5, 1))
```

**Figure S4**. Microbial community composition based on Class occurrences.<br>

```{r rel abund barplot genus, message=F, echo=T, eval=T, warning=T, include=T, cache=F, fig.width=10, fig.height=4}

# input data
ps_Genus <- psmelt(transform_sample_counts(tax_glom(psdata_RBRA, "Genus"), fun = function(x) x/sum(x)))

# relative abundance based on Genus
genus_count <- ps_Genus %>% 
  as_tibble() %>% 
  mutate(Genus = factor(Genus)) %>% 
  pull(Genus) %>% 
  levels() # count genera n = 1041

Genus_abundances <- ps_Genus %>% 
  as_tibble() %>% 
  dplyr::select(sampleid, redox_type, sample_type, Genus, Abundance) %>% 
  mutate(redox_type = as.character(redox_type),
         sample_type = as.character(sample_type)) %>% 
  group_by(sampleid, redox_type, sample_type, Genus) %>% 
  summarize(rel_abund = sum(Abundance), .groups = "drop") %>% 
  group_by(redox_type,sample_type, Genus) %>% 
  summarize(mean_rel_abund = 100* mean(rel_abund), .groups = "drop") %>% 
  mutate(Genus = str_replace(Genus, "(.*)_unclassified", "Unclassified *\\1*"),
         Genus = str_replace(Genus, "^(\\S*)$", "*\\1*"))

Genus_pool <- Genus_abundances %>% 
  group_by(Genus) %>% 
  summarize(pool = max(mean_rel_abund) < 8, 
            mean = mean(mean_rel_abund), 
            .groups = "drop")

# plot relative abundance based on Class -> 21 most abundant genus
plot_Genus <- inner_join(Genus_abundances, Genus_pool, by="Genus") %>% 
  mutate(Genus = if_else(pool, "Other", Genus)) %>% 
  group_by(redox_type, sample_type, Genus) %>%
  summarize(mean_rel_abund = sum(mean_rel_abund), 
            mean = min(mean),
            .groups = "drop") %>% 
  mutate(Genus = factor(Genus), 
         Genus = fct_reorder(Genus, mean, .desc = T),
         redox_type = factor(redox_type),
         redox_type = fct_relevel(redox_type, rev(c("methanogenic","sulphate", "iron", "nitrate", "aerobic", "inoculum"))),
         sample_type = factor(sample_type),
         sample_type = fct_relevel(sample_type, rev(c("ind", "ditch", "mun", "soil")))) %>% 
ggplot(aes(x=redox_type, y = mean_rel_abund, fill = forcats::fct_rev(Genus))) +
  geom_col() +
  scale_fill_manual(name=NULL, 
                    values = c("white", "#5AAE61", "#1B7837", "#FEE090", "#FDAE61", "#9970AB", "#762A83", "#7FBC41", "#4D9221", "#DE77AE", "#C51B7D", "#BF812D", "#8C510A", "#E6F598", "#ABDDA4", "#35978F", "#01665E", "#878787", "#4D4D4D", "#4393C3", "#2166AC", "#D6604D", "#B2182B")) +
  facet_wrap(~sample_type, ncol = 4) +
  scale_y_continuous(expand = c(0,0)) +
  labs(x="Redox condition", y="Mean Relative Abundance (%)") +
  theme_classic() +
  theme(axis.text = element_text(size = 7, colour = "black"),
        axis.ticks.x = element_line(size = 0.35),
        axis.ticks.y = element_line(size = 0.35),
        axis.text.x	= element_text(angle = 45, hjust = 1),
        axis.title = element_text(size = 9),
        axis.title.x = element_text(margin = margin(t = 7)),
        axis.title.y = element_text(margin = margin(r = 7)),
        axis.line = element_blank(),
        legend.text = element_markdown(size=7),
        legend.title = element_text(size = 8),
        legend.key.size = unit(10, "pt"),
        strip.text = element_text(size = 8),
        strip.background = element_blank(),
        panel.spacing = unit(10, "pt"),
        panel.border = element_rect(colour = "black", 
                                    fill = NA, 
                                    size = 0.5))

genus = plot_grid(plot_Genus + theme(legend.position = "none"), 
                   ncol = 1, 
                   nrow = 1, 
                   align = "hv")

legend_bar <- get_legend(plot_Genus + guides(fill = guide_legend(ncol = 1)) +
    theme(legend.position = "right"))

plot_grid(genus, legend_bar, ncol=3, rel_widths = c(2, 0.5, 0))
```

**Figure S5**. Microbial community composition based on Genus occurrences â€“ 21 most abundant genus.<br>